didx-xyz/yoma-mobile
# YOMA-606 - feat(OUTH): SSO setup

## Scope [YOMA-606](https://yomaworld.atlassian.net/browse/YOMA-606)

![](https://media.giphy.com/media/l3vRhDQrmQ6Rj4Y4E/giphy.gif)
## Work Done
(A LOT)
- adding SSO/OAUTH connection for authorizing the user.
- removed all unused auth flows/screens (login, register, etc)
- adding new api calls to get missing user id
- updating api config to be able to take the `api/v1` as a separate property, to enable calling the `connect` endpoint (That doesn't have the api/v1 part.
- updating User credential calls to only happen once we have the user id in state, moved this to a hydrateUser middleware that calls all actions dependant on the userID once we have it in state.
- updated the Landing page to only have a Login button.

## Steps to Test
try logging in - should open a webview, then ensure that all the user's data is loaded as usual.
    
## Screenshots
<img width="444" alt="Screenshot 2022-04-16 at 22 40 09" src="https://user-images.githubusercontent.com/1424561/163690714-c20ecd68-8c76-46f5-88a1-40eae9c0458e.png">
